Jellyfin.Server/Extensions/ApiServiceCollectionExtensions.cs

```diff
@@ -169,36 +171,104 @@ public static AuthenticationBuilder AddCustomAuthentication(this IServiceCollect
-        /// <param name="knownProxies">A list of all known proxies to trust for X-Forwarded-For.</param>
+        /// <param name="config">The <see cref="NetworkConfiguration"/>.</param>
         /// <returns>The MVC builder.</returns>
-        public static IMvcBuilder AddJellyfinApi(this IServiceCollection serviceCollection, IEnumerable<Assembly> pluginAssemblies, IReadOnlyList<string> knownProxies)
+        public static IMvcBuilder AddJellyfinApi(this IServiceCollection serviceCollection, IEnumerable<Assembly> pluginAssemblies, NetworkConfiguration config)
         {
             IMvcBuilder mvcBuilder = serviceCollection
-                .AddCors()
-                .AddTransient<ICorsPolicyProvider, CorsPolicyProvider>()
-                .Configure<ForwardedHeadersOptions>(options =>
+            .AddCors()
+            .AddTransient<ICorsPolicyProvider, CorsPolicyProvider>()
+            .Configure<ForwardedHeadersOptions>(options =>
                 {
+                    // https://github.com/dotnet/aspnetcore/blob/master/src/Middleware/HttpOverrides/src/ForwardedHeadersMiddleware.cs
+                    // Enable debug logging on Microsoft.AspNetCore.HttpOverrides.ForwardedHeadersMiddleware to help investigate issues.
+
                     options.ForwardedHeaders = ForwardedHeaders.XForwardedFor | ForwardedHeaders.XForwardedProto;
-                    if (knownProxies.Count == 0)
+                    if (config.KnownProxies.Length == 0)
                     {
-                        options.KnownNetworks.Clear();
                         options.KnownProxies.Clear();
                     }
                     else
                     {
-                        for (var i = 0; i < knownProxies.Count; i++)
+                        ParseList(config, config.KnownProxies, delegate(IPAddress address, int _)
                         {
-                            if (IPHost.TryParse(knownProxies[i], out var host))
-                            {
-                                options.KnownProxies.Add(host.Address);
-                            }
-                        }
+                            options.KnownProxies.Add(address);
+                        });
+                    }
+
+                    if (config.KnownNetworks.Length == 0)
+                    {
+                        options.KnownNetworks.Clear();
                     }
+                    else
+                    {
+                        ParseList(config, config.KnownNetworks, delegate(IPAddress address, int prefix)
+                        {
+                            options.KnownNetworks.Add(new IPNetwork(address, prefix));
+                        });
+                    }
+
+                    // Only set forward limit if we have some known proxies or some known networks.
+                    options.ForwardLimit = options.KnownProxies.Count + options.KnownNetworks.Count == 0 ? 1 : null;
```

Comment:
```suggestion
                    if (options.KnownProxies.Count != 0)
                    {
                        options.ForwardLimit = null;
                    }
```

Might as well use the actual default in case it changes